src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/events/OutboundEventsPublisher.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.digital.hmpps.whereabouts.services.court.events
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+import software.amazon.awssdk.services.sns.model.PublishRequest
+import uk.gov.justice.digital.hmpps.whereabouts.config.Feature
+import uk.gov.justice.digital.hmpps.whereabouts.config.FeatureSwitches
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+
+@Service
+class OutboundEventsPublisher(
+  private val hmppsQueueService: HmppsQueueService,
+  private val mapper: ObjectMapper,
+  features: FeatureSwitches,
+) {
+  private val outboundEventsEnabled = features.isEnabled(Feature.OUTBOUND_EVENTS_ENABLED)
+
+  companion object {
+    const val TOPIC_ID = "domainevents"
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  init {
+    log.info("Outbound SNS event publishing enabled = $outboundEventsEnabled")
+  }
+
+  private val domainEventsTopic by lazy {
+    hmppsQueueService.findByTopicId(TOPIC_ID) ?: throw RuntimeException("Topic with name $TOPIC_ID doesn't exist")
+  }
+
+  fun send(event: OutboundHMPPSDomainEvent) {
+    if (outboundEventsEnabled) {
+      domainEventsTopic.snsClient.publish(
```

Comment:
I'd suggest looking at our service this publish isn't quite right.  The is a result of the old example used in the SQS/SNS example project.  This might mean updating to a later version of the SQS plugin though ...

Reply:
It was taken from A&A - and does work with the HMPPS SQS lib at v4.0.1. It does appear to produce the same event format as per the later one in BVLS API, albeit not catching exceptions on the send. How important do you think it is to update to the latest?

Reply:
See example [here](https://github.com/ministryofjustice/hmpps-book-a-video-link-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/OutboundEventsPublisher.kt#L40), is the publish bit. A subtle difference.